### DISCUSSION TITLE
Component Removable Method

### DISCUSSION THREAD

**Discussion by RakulAgn** [GrapesJS/grapesjs#5578] [General]
Whenever a component is set to `removable: false`, the toolbar correctly removes the delete option. However, there is another way to remove the component, even when it is set as non-removable: by using the Undo Manager, which can remove the block. I'm not sure if this is the correct approach, but one possible solution is to clear the undo stack when dropping those types of blocks.

**Comment by artf** [comment]
Yes, the Undo Manager definitely shouldn't make any difference if the component is removable or not, otherwise, it would lead to a mess (think about undoing a component containing mixed non/removable components).  Clearing the undo stack is pretty easy via [editor.UndoManager.clear()](https://grapesjs.com/docs/api/undo_manager.html#clear)
